### ISSUE TITLE
False positive `()` doesn't implement `Display` (required by {})

### ISSUE THREAD

**Issue by neonaot** [intellij-rust/intellij-rust#8909]
## Environment

* **IntelliJ Rust plugin version:** 0.4.172.4663-222-nightly
* **Rust toolchain version:** 1.63.0-nightly (76761db59 2022-05-24) x86_64-apple-darwin
* **IDE name and version:** CLion 2022.2 EAP (CL-222.2680.4)
* **Operating system:** macOS 11.4
* **Macro expansion engine:** new
* **Name resolution engine:** new
* **Additional experimental features:** org.rust.cargo.features.settings.gutter, org.rust.cargo.emulate.terminal, org.rust.cargo.evaluate.build.scripts, org.rust.macros.proc

Originally posted in youtrack https://youtrack.jetbrains.com/issue/CPP-29450/Intellij-IDEA-rust-plugin-displaying-error-that-doesnt-exist

## Problem description

<img width="589" alt="изображение" src="https://user-images.githubusercontent.com/49211026/172157834-ace729a6-fc1e-409a-9eb1-0be986a07ba6.png">

## Steps to reproduce
```rust
fn main() {
    let factor = 3;
    let multiply = | a | a * factor;
    println!("{}", multiply(13));
}
```

**Comment by Undin** [comment]
Duplicate of #8892
